### PR TITLE
Review tags

### PR DESCRIPTION
**Pull request by adescoteaux1**
## Description

[Link to Ticket](https://github.com/GenerateNU/platnm/issues/105#issue-2600788215)

- Created a endpoint to display all tags
- updated create review so that tags can be passed in as well and are added to the review_tags file
- throws an error if the tag does not exist

## How Has This Been Tested?
GetTags:
image

Updating create review:
Pasted Graphic
Review_Tag Before:
RLS disabled
Revew_tag After:
Y Filter

Error:
Pasted Graphic 3

Adding multiple Tags:
Scripts
Pasted Graphic 5

No Change without tag:
image
image
Still added to db:
image

## Checklist:
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] Some tests are included that test my code

### CODE REVIEW

**Comment by adescoteaux1** [comment]
Having the same issue as previous PR where i can't fix that lint issue and npm run format is not a command that I have

**Comment by adescoteaux1** [comment]
Also Realized that if the tag is not a valid tag, the review is still added but the tag is not. Is that the right logic or do we want to not create the review as well. I don't really see this path being run at all since tags will be selected from a preset list anyway, but just a thought.

**Comment by ddusichka** [comment]
I agree with you about the invalid tag value but review still gets created case! since the frontend should ideally only ever pass real tags, we should be fine. and even if it did, I'd rather have the review get made and be missing a tag than make the user redo the whole request.

also, for the lint issues, I think you just need to delete the searchAndHandleSpotifyMedia method that's leftover in the spotify handler package (since it's duplicated in the media package as well). 
(side note npm run format is actually not what you need here (it fixes the frontend only), but it should work if you cd into frontend? not needed here rn though)
image

File: backend/internal/service/handler/media/get_media.go [hunk 1]
```diff
@@ -23,6 +27,24 @@ func (h *Handler) GetMediaByName(c *fiber.Ctx) error {
 		mediaType = models.BothMedia
 	}
 	medias, _ := h.mediaRepository.GetMediaByName(c.Context(), name, mediaType)
+	log.Println(medias)
```
Comment (ddusichka) [inline backend/internal/service/handler/media/get_media.go:30]: i think changes from your other branch snuck into here :( can u remove them for the time being while we figure out that other PR?

File: backend/internal/service/server.go [hunk 2]
```diff
@@ -65,6 +65,7 @@ func setupRoutes(app *fiber.App, config config.Config) {
 	app.Route("/reviews", func(r fiber.Router) {
 		reviewHandler := reviews.NewHandler(repository.Review, repository.User, repository.UserReviewVote)
 		r.Post("/", reviewHandler.CreateReview)
+		r.Get("/tag", reviewHandler.GetTags)
```
Comment (ddusichka) [inline backend/internal/service/server.go:68]: super nit sorry but can we call the endpoint `/tags`

File: backend/internal/storage/postgres/schema/review.go [hunk 3]
```diff
@@ -67,6 +67,28 @@ func (r *ReviewRepository) CreateReview(ctx context.Context, review *models.Revi
...
 
+	// Fetch or create tag IDs and add them to the review_tags table
+	for _, label := range review.Tags {
+		var tagID int
+
+		// Check if the tag already exists
+		tagQuery := `SELECT id FROM tag WHERE name = $1`
+		err := r.QueryRow(ctx, tagQuery, label).Scan(&tagID)
+
+		// If the tag doesn't exist, insert it and retrieve its ID
```
Comment (ddusichka) [inline backend/internal/storage/postgres/schema/review.go:78]: we should prob delete/update this comment since it doesn't match the actual behavior

**Review by ddusichka — COMMENTED** [review]
tag logic looks good!! just tiny suggestions + removing the search_spotify stuff if possible

File: backend/internal/service/server.go (hunk 2 above)
Comment (adescoteaux1) [reply backend/internal/service/server.go]: fixed!

File: backend/internal/service/handler/media/get_media.go (hunk 1 above)
Comment (adescoteaux1) [reply backend/internal/service/handler/media/get_media.go]: very true, removed!

**Review by ddusichka — APPROVED** [review]
yay tags!

**Review by ddusichka — APPROVED** [review]

**Pull request merged**